content/blog/2024/06/kubewarden-1-13-release.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Kubewarden 1.13 release
+authors:
+  - Flavio Castelli
+date: 2024-06-05
+---
+
+I'm pleased to announce a new release of Kubewarden, version 1.13. This release features a series of
+improvements and bug fixes that contribute to better performance and stability.
+
+Let's go through the most significant changes.
+
+## Policy Server memory usage
+
+A community member reported that the Kubewarden Policy Server was using a lot of memory, especially
+when running context aware policies on big clusters. The number of resources being accessed by the
+policies was significantly high, in the order of 3200 Namespaces, 10500 Ingresses, 200 ClusterRoleBindings
+and 11000 RoleBindings.
+Under certain circumstances, the amount of memory being used caused the Policy Server to be terminated by the kernel's OOM killer.
+
+After some investigation, we found two different root causes of this issue.
+
+The first one was related to [kube-rs](https://github.com/kube-rs), the Rust library we use to interact with Kubernetes.
+The library was doing excessive allocations when building the initial state of a resource. We fixed this issue
+with [this PR](https://github.com/kube-rs/kube/pull/1494).
+With this fix we reduced the peak of memory usage caused by the creation of the initial state of a Kubernetes resource.
+
+The second issue was related with how the memory allocator of Rust works. By default, Rust uses the system allocator (glibc malloc in our case).
+It turns out the system allocator is not releasing the allocated memory back to the system when this is no longer needed.
+We changed Policy Server to use the [jemalloc](http://jemalloc.net/) allocator which, on top of providing great performances, it has many
+configuration knobs.
+After some experimentations, we found the right balance between performances and memory usage.
```

Comment:
experimentation or experiments and singular performance

Reply:
done, thanks